d3a_api_client/commands.py

```diff
@@ -14,39 +15,47 @@ def buffer_length(self):
         return len(self._commands_buffer)
 
     def offer_energy(
-            self, area_uuid: str, energy: float, price: float, replace_existing: bool = True):
+            self, area_uuid: str, energy: float, price: float, replace_existing: bool = True,
+            attributes: Dict = None, requirements: List[Dict] = None):
 
         return self._add_to_buffer(
             area_uuid,
             Commands.OFFER,
-            {'energy': energy, 'price': price, 'replace_existing': replace_existing})
+            {"energy": energy, "price": price, "replace_existing": replace_existing,
+             "attributes": attributes, "requirements": requirements})
 
     def offer_energy_rate(
-            self, area_uuid: str, energy: float, rate: float, replace_existing: bool = True):
+            self, area_uuid: str, energy: float, rate: float, replace_existing: bool = True,
+            attributes: dict = None, requirements: List[Dict] = None):
```

Comment:
```suggestion
            attributes: Dict = None, requirements: List[Dict] = None):
```
The same applies to the rest of the PR.